server/src/main/java/org/elasticsearch/action/admin/cluster/node/hotthreads/TransportNodesHotThreadsAction.java

```diff
@@ -32,21 +34,17 @@ public class TransportNodesHotThreadsAction extends TransportNodesAction<
     TransportNodesHotThreadsAction.NodeRequest,
     NodeHotThreads> {
 
+    private static final String NAME = "cluster:monitor/nodes/hot_threads";
```

Comment:
Could we inline the action name in the `ActionType` constructor and then use `TransportNodesHotThreadsAction.TYPE.name()` wherever it's needed? That way everything is going through the `ActionType` instance which ties together the transport action with the client-facing action better than just using a bare string.

Reply:
Sure thing, did it for all but one of them because there the name went into a switch statement.